src/launcher/java/org/truffleruby/launcher/RubyLauncher.java

```diff
@@ -186,7 +185,14 @@ protected void printHelp(OptionCategory maxCategory) {
 
     @Override
     protected AbortException abortUnrecognizedArgument(String argument) {
-        throw abortInvalidArgument(argument, "truffleruby: invalid option " + argument + "  (Use --help for usage instructions.)");
+        final String description;
+        if (argument.startsWith("--ruby.")) {
+            description = argument + " (-X" + argument.substring(7) + ")";
+        } else {
+            description = argument;
+        }
+
+        throw abortInvalidArgument(argument, "truffleruby: invalid option " + description + "  (Use --help for usage instructions.)");
```

Comment:
Tracking the original argument is not something which can be easily done. I think this is a good compromise instead of duplicating the option parsing code. Also since we now have GraalVM launcher the `--ruby.` should be the default option form, keeping `-X` as rather legacy alias. The error message somewhat matches it. The GraalVM launcher also allows to drop the `ruby.` prefix when in Ruby launcher so `jt ruby --show_version=true -e p:a` works as `jt ruby --ruby.show_version=true -e p:a`.

Reply:
It's not ideal, but I can see it deduplicates code.
Do you have an example of a bug fixed by this?

Reply:
Without the unification some polyglot options depending on the source could bypass the code which was suppose to handle it and the option was ignored, e.g. `--ruby.show_version=true --ruby.to_execute=p:b --ruby.execution_action=INLINE` was not working.

Reply:
What do you mean by "polyglot options depending on the source"?

Reply:
polyglot options = `--ruby....`, source = -X or --ruby form.